README.md

```diff
@@ -1,9 +1,42 @@
 <!-- default badges list -->
-![](https://img.shields.io/endpoint?url=https://codecentral.devexpress.com/api/v1/VersionRange/587734441/22.2.3%2B)
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T1145991)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
 # XAF - Customize Logon Parameters
 
+This example demonstrates how to customize standard authentication behavior and supply additional logon parameters in XAF applications. The default login form displays **User Name** and **Password** editors. This sample customizes the login form to show **Company** and **Application User** lookup editors instead of **User Name**.
 
-Under constuction.
+![Login form](https://docs.devexpress.com/eXpressAppFramework/images/custom-logon-prameters-blazor.png)
+
+## Implementation Details
+
+### ASP.NET Core Blazor
+
+1. Define custom parameters. To do this, add a new class with parameters to your project. In this example, this is the [Company](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/Company.cs) class. This class should contain company names and a list of `ApplicationUser` objects as a part of a one-to-many relationship. Then, add the second part of this relationship to the [ApplicationUser](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/ApplicationUser.cs#L28) class generated by the Solution Wizard. And finally, add the `Company` class to your application’s [DbContext](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/EFCoreCustomLogonAllDbContext.cs#L43).
+2. Add custom parameters to the login window. For an example, refer to the following file - [CustomLogonParameters.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomLogonParameters.cs).
+3. Implement [custom authentication](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthentication.cs).
+4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs).
+5. Pass custom classes to the Security System. For an example, refer to the following file - [Startup.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Startup.cs#L73-L78).
+6. Add demo data.
```

Comment:
link to https://github.com/DevExpress-Examples/xaf-custom-logon-parameters/blob/update-readme/CS/EF/EFCoreCustomLogonAll.Module/DatabaseUpdate/Updater.cs

Reply:
```suggestion
6. Add [demo data](./CS/EF/EFCoreCustomLogonAll.Module/DatabaseUpdate/Updater.cs).
```